### PR TITLE
Add support for passing SQL variables as query parameters

### LINKED ISSUE #61: Add support for date parameter for `analytics` queries
Example: http://localhost/analytics/v1/podcastGender?date=2023-02-12

### PR DESCRIPTION
**Pull request by mre**
Fixes #61.

### CODE REVIEW

**Comment by woolfg** [comment]
@mre can you pls have a look

File: src/db/AnalyticsRepository.ts [hunk 1]
```diff
@@ -14,18 +14,35 @@ class AnalyticsRepository {
...
-    async execute(endpoint: string): Promise<any> {
+    async execute(endpoint: string, sqlVars = {}): Promise<any> {
         // Look up the query text for the endpoint
         const query = this.queries.get(endpoint)
 
         if (!query) {
             throw new Error(`Query not found for endpoint ${endpoint}`)
         }
 
+        // Iterate over the  and escape them
```
Comment (mre) [inline src/db/AnalyticsRepository.ts:25]:
```suggestion
        // Iterate over the values  and escape them
```

File: src/db/AnalyticsRepository.ts [hunk 2]
```diff
@@ -14,18 +14,35 @@ class AnalyticsRepository {
...
-    async execute(endpoint: string): Promise<any> {
+    async execute(endpoint: string, sqlVars = {}): Promise<any> {
         // Look up the query text for the endpoint
         const query = this.queries.get(endpoint)
 
         if (!query) {
             throw new Error(`Query not found for endpoint ${endpoint}`)
         }
 
+        // Iterate over the values  and escape them
```
Comment (mre) [inline src/db/AnalyticsRepository.ts:25]:
```suggestion
        // Iterate over the values and escape them
```

**Comment by mre** [comment]
Looks great. I didn't run it yet, but I couldn't spot any issues.

**Pull request merged**